Schemata/sarif-schema.json

```diff
@@ -353,6 +272,54 @@
       }
     },
 
+    "externalFiles": {
+      "description": "References to external files that should be inlined with the content of a root log file.",
+      "additionalProperties": false,
+      "type": "object",
+      "properties": {
+
+        "conversion": {
+          "description": "An external reference to a run.conversion object to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
+        },
+
+        "files": {
+          "description": "An external reference to a run.files object to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
+        },
+
+        "graphs": {
+          "description": "An external reference to a run.graphs object to be merged with the root log file.",
+          "$ref": "#/definitions/fileLocation"
```

Comment:
>"$ref": "#/definitions/fileLocation" [](start = 10, length = 36)

`run.graphs` is actually an array of `graph` objects. So the value of this property must be an array of `fileLocation` objects.

BUT! I just realized that there is a bug in the spec. `run.graphs` (and `result.graphs`, for that matter) should _not_ be an array. It should be a dictionary whose key is `graph.id` (just like in the `resources.rules` dictionary the key is `rule.id`).

This matters because we have `graphTraversal` objects which refer to the `graph` they traverse by specifying the graph's id. So we should be able to look up the referenced graph in a dictionary. I will file a bug.

Bottom line: Leave this property alone (_don't_ make it an array) because we'll fix up the `run.graphs` property (and `result.graphs`) to be a dictionary instead of an array.

Reply:
I filed [Issue #237]([https://github.com/oasis-tcs/sarif-spec/issues/237](https://github.com/oasis-tcs/sarif-spec/issues/237), ), "run.graphs and result.graphs should be dictionaries, not arrays."

---
In reply to: [215688773](https://github.com/oasis-tcs/sarif-spec/pull/236#discussion_r215688773) [](ancestors = 215688773)

Reply:
I'm sorry, I now regret my "bottom line" comment above. As the spec stands, `run.graphs` is an array, and so I revert to my previous position: The value of this property must be an array of `fileLocation` objects. On the day we fix #237, we will change this as well.

---
In reply to: [215688773](https://github.com/oasis-tcs/sarif-spec/pull/236#discussion_r215688773) [](ancestors = 215688773)